plone/formwidget/namedfile/widget.py

```diff
@@ -190,6 +237,36 @@ def extract(self, default=NOVALUE):
         elif action == 'nochange':
             if self.value is not None:
                 return self.value
+
+            # Handle already uploaded files in case of previous form errors
+            file_upload_id = self.request.get(
+                "%s.file_upload_id" % self.name, None
+            )
+            if file_upload_id:
+                upload_map = getattr(
+                    self.context.temp_folder.uploads,
+                    'upload_map',
+                    {}
+                )
+                fileinfo = upload_map.get(file_upload_id, {})
+                filename = fileinfo.get('filename')
+                data = fileinfo.get('data')
+                if filename or data:
+                    filename = safe_basename(filename)
+                    if (
+                            filename is not None
+                            and not isinstance(filename, unicode)
+                    ):
+                        # work-around for
```

Comment:
This is partly copied from: https://github.com/plone/plone.formwidget.namedfile/blob/a3f996c41cf288e1f8e90a279f20d7c4d42155ec/plone/formwidget/namedfile/converter.py#L35